lecture_3/03_project/calculator/internal/infrastructure/sqlite/betRepository.go

```diff
@@ -0,0 +1,186 @@
+package sqlite
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/pkg/errors"
+
+	domainmodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/domain/models"
+	storagemodels "github.com/superbet-group/code-cadets-2021/lecture_3/03_project/calculator/internal/infrastructure/sqlite/models"
+)
+
+// BetRepository provides methods that operate on cals_bets SQLite database.
+type BetRepository struct {
+	dbExecutor DatabaseExecutor
+	betMapper  BetMapper
+}
+
+// NewBetRepository creates and returns a new BetRepository.
+func NewBetRepository(dbExecutor DatabaseExecutor, betMapper BetMapper) *BetRepository {
+	return &BetRepository{
+		dbExecutor: dbExecutor,
+		betMapper:  betMapper,
+	}
+}
+
+// InsertBetCalculated inserts the provided calculated bet into the database. An error is returned if the operation
+// has failed.
+func (r *BetRepository) InsertBetCalculated(ctx context.Context, bet domainmodels.BetCalculated) error {
+	storageBet := r.betMapper.MapDomainBetToStorageBet(bet)
+	err := r.queryInsertBet(ctx, storageBet)
+	if err != nil {
+		return errors.Wrap(err, "calc_bets repository failed to insert a bet with id "+bet.Id)
```

Comment:
It's up to you what you want to put in the error message, but I just want to point out I've used `"bet repository failed to..." `text in Controller because this struct is called `BetRepository`, not because it's operating on `bets` database. We usually use that convention in our production code, it is helpful for debugging to have the struct name within the error message.